zivid_camera/test/test_zivid_camera.cpp

```diff
@@ -239,79 +246,204 @@ TEST_F(ZividNodeTest, testCapturePublishesTopics)
+  ASSERT_EQ(point_cloud->point_step, 16U);  // 3x4 bytes for xyz + 4 bytes padding (w)
+  ASSERT_EQ(point_cloud->row_step, point_cloud->width * point_cloud->point_step);
+  ASSERT_EQ(point_cloud->is_dense, false);
+  ASSERT_EQ(point_cloud->data.size(), point_cloud->width * point_cloud->height * point_cloud->point_step);
+
+  auto pointCloud = captureViaSDKDefaultSettings();
+  auto expectedXYZ = pointCloud.copyData<Zivid::PointXYZ>();
+  ASSERT_EQ(point_cloud->width, expectedXYZ.width());
+  ASSERT_EQ(point_cloud->height, expectedXYZ.height());
+  for (std::size_t i = 0; i < expectedXYZ.size(); i++)
+  {
+    uint8_t* point_ptr = &point_cloud->data[i * point_cloud->point_step];
+    const float x = *reinterpret_cast<float*>(&point_ptr[0]);
+    const float y = *reinterpret_cast<float*>(&point_ptr[4]);
+    const float z = *reinterpret_cast<float*>(&point_ptr[8]);
+
+    const auto expected = expectedXYZ(i);
+    comparePointCoordinate(x, expected.x);
+    comparePointCoordinate(y, expected.y);
+    comparePointCoordinate(z, expected.z);
+  }
+}
 
+TEST_F(CaptureOutputTest, testCapture3DColorImage)
+{
   std::optional<sensor_msgs::Image> image;
   auto color_image_sub =
       subscribe<sensor_msgs::Image>(color_image_color_topic_name, [&](const auto& i) { image = *i; });
-  enableFirst3DFrame();
-  zivid_camera::Capture capture;
-  ASSERT_TRUE(ros::service::call(capture_service_name, capture));
-  short_wait_duration.sleep();
+
+  enableFirst3DAcquisitionAndCapture();
+
   ASSERT_TRUE(image.has_value());
   ASSERT_EQ(image->width, 1920U);
   ASSERT_EQ(image->height, 1200U);
-  constexpr uint32_t bytes_per_pixel = 3U;
-  ASSERT_EQ(image->step, bytes_per_pixel * 1920U);
+  constexpr uint32_t bytes_per_pixel = 4U;
+  ASSERT_EQ(image->step, bytes_per_pixel * image->width);
   ASSERT_EQ(image->data.size(), image->step * image->height);
-  ASSERT_EQ(image->encoding, "rgb8");
+  ASSERT_EQ(image->encoding, "rgba8");
   ASSERT_EQ(image->is_bigendian, false);
 
-  auto verifyPixelColor = [&](const FileCameraExpectedRGB& expectedRGB) {
-    const auto index = expectedRGB.row * image->step + 3 * expectedRGB.col;
-    ASSERT_EQ(image->data[index], expectedRGB.r);
-    ASSERT_EQ(image->data[index + 1], expectedRGB.g);
-    ASSERT_EQ(image->data[index + 2], expectedRGB.b);
-  };
+  const auto pointCloud = captureViaSDKDefaultSettings();
+  const auto expectedRGBA = pointCloud.copyData<Zivid::ColorRGBA>();
+  ASSERT_EQ(image->width, expectedRGBA.width());
+  ASSERT_EQ(image->height, expectedRGBA.height());
+  for (std::size_t i = 0; i < expectedRGBA.size(); i++)
+  {
+    const auto expected = expectedRGBA(i);
+    const auto index = i * bytes_per_pixel;
+    ASSERT_EQ(image->data[index], expected.r);
+    ASSERT_EQ(image->data[index + 1], expected.g);
+    ASSERT_EQ(image->data[index + 2], expected.b);
+    ASSERT_EQ(image->data[index + 3], expected.a);
+    ASSERT_EQ(expected.a, 255);
+  }
+}
+
+TEST_F(CaptureOutputTest, testCaptureDepthImage)
```

Comment:
Testing the snr and testing the depth looks very similar except the topic name and type T in `copyData<T>()`.
Perhaps we could re-use?

Reply:
Closing this due to the other comment

> For points we have converted the output from SDK (mm) to (m) so when we compare we divide SDK output by 1000 and compare it to ROS output.